### PR TITLE
add "latin1" encoding support for lightning

### PR DESCRIPTION
**Pull request by lance6716**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/tidb-lightning-configuration.md [hunk 1]
```diff
@@ -205,6 +205,7 @@ data-source-dir = "/data/my_database"
 #  - utf8mb4：表结构文件必须使用 UTF-8 编码，否则会报错。
 #  - gb18030：表结构文件必须使用 GB-18030 编码，否则会报错。
 #  - auto：自动判断文件编码是 UTF-8 还是 GB-18030，两者皆非则会报错（默认）。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-configuration.md:207]: auto 这行是否也需要加上 MySQL latin1

File: tidb-lightning/tidb-lightning-configuration.md [hunk 2]
```diff
@@ -205,6 +205,7 @@ data-source-dir = "/data/my_database"
...
+#  - latin1：源数据文件使用 MySQL latin1 字符集编码（也被称为 Code Page 1252）。
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-configuration.md:208]: ChatGPT 说 
“在 MySQL 中，latin1 字符集表示 ISO 8859-1 字符集，而 Code Page 1252 是 Windows 操作系统上的字符编码。虽然这两种字符集在许多字符方面是相似的，但在某些字符上有细微的差异。
因此，如果您在 MySQL 中使用 latin1 字符集，它实际上代表 ISO 8859-1 字符集，而不是 Code Page 1252。请注意，这两个术语指代不同的字符集编码。”

Comment (lance6716) [reply tidb-lightning/tidb-lightning-configuration.md:208]: 事实错误

Comment (lance6716) [reply tidb-lightning/tidb-lightning-configuration.md:208]: https://dev.mysql.com/doc/refman/8.0/en/charset-we-sets.html

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by D3Hunter — APPROVED** [review]

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**